pkg/node/stager.go

```diff
@@ -165,15 +168,31 @@ func (n *NFSStager) Stage(ctx context.Context, req *csi.NodeStageVolumeRequest,
 	}
 
 	// Create folder with 1777 in nfs share so every user can use it
-	log.WithFields(logFields).Info("creating common folder")
 	if err := fs.MkdirAll(filepath.Join(stagingPath, commonNfsVolumeFolder), 0750); err != nil {
 		return nil, status.Errorf(codes.Internal,
 			"can't create common folder %s: %s", filepath.Join(stagingPath, "volume"), err.Error())
 	}
 
-	if err := fs.Chmod(filepath.Join(stagingPath, commonNfsVolumeFolder), os.ModeSticky|os.ModePerm); err != nil {
-		return nil, status.Errorf(codes.Internal,
-			"can't change permissions of folder %s: %s", filepath.Join(stagingPath, "volume"), err.Error())
+	mode := os.ModePerm
+	acls := req.PublishContext[common.KeyNfsACL]
+	aclsConfigured := false
+	if acls != "" {
+		if posixMode(acls) {
+			perm, _ := strconv.ParseUint(acls, 8, 64)
```

Comment:
Please handle the error while parsing, otherwise it may cause program crash in case of any failure in parsing

Reply:
done